Content.Server/Chemistry/Components/InjectorComponent.cs

```diff
@@ -163,6 +186,51 @@ async Task<bool> IAfterInteract.AfterInteract(AfterInteractEventArgs eventArgs)
             return true;
         }
 
+        /// <summary>
+        /// Send informative pop-up messages and wait for a Do-After to complete.
+        /// </summary>
+        public async Task<bool> TryInjectDoAfter(EntityUid user, EntityUid target)
+        {
+            InUse = true;
+            var popupSys = EntitySystem.Get<SharedPopupSystem>();
+
+            // for the user
+            popupSys.PopupEntity(Loc.GetString("injector-component-injecting-user"), target, Filter.Entities(user));
+
+            if (user != target)
+            {
+                // for the target
+                var userName = Owner.EntityManager.GetComponent<MetaDataComponent>(user).EntityName;
+                popupSys.PopupEntity(Loc.GetString("injector-component-injecting-target",
+                    ("user", userName)), user, Filter.Entities(target));
+
+                // also, lets log this
+                var userEntity = Owner.EntityManager.GetEntity(user);
+                var targetEntity = Owner.EntityManager.GetEntity(target);
+
+                var text = ToggleState == InjectorToggleMode.Inject
+                    ? "inject reagents into"
+                    : "draw reagents from";
+
+                // using the "force feed" log type. It's not quite feeding, but the effect is the same.
+                EntitySystem.Get<AdminLogSystem>().Add(LogType.ForceFeed,
+                    $"{userEntity} is attempting to {text} {targetEntity}");
+            }
+
+            var status = await EntitySystem.Get<DoAfterSystem>().WaitDoAfter(
+                new DoAfterEventArgs(user, Delay, target: target)
+                {
+                    BreakOnUserMove = true,
+                    BreakOnDamage = true,
```

Comment:
If this delay also happens for self injections, then this might need to be changed if trying to heal self when poisoned.

Reply:
different sources of damage can specify whether to they should break doafters, so this isn't an issue

Reply:
Yeah there used to be issues with do-after damage breaking, like people being unable to crowbar doors because they were suffocating.

Now poison, suffocation and other non-combat related damage **shouldn't** break do-afters.